README
NAME
    FLTK - Perl bindings to the Fast Light Toolkit

Synopsis
    use strict;
    use warnings;
    use FLTK qw[:style];

    my $win = FLTK::Window->new(300, 180);
    my $box = $win->add(FLTK::Widget->new(20, 40, 260, 100, 'Hello, World!'));
    $box->box(UP_BOX);
    $box->labelfont(HELVETICA_BOLD_ITALIC);
    $box->labelsize(36);
    $box->labeltype(SHADOW_LABEL);
    $win->show();
    exit FLTK::run();

Description
    FLTK is a graphical user interface toolkit for X (UNIX®), Microsoft®
    Windows®, OS/X, and several other platforms. FLTK provides modern GUI
    functionality without the bloat and supports 3D graphics via OpenGL® and
    its built-in GLUT emulation.

See Also
    FLTK::Notes, FLTK::Basics, FLTK::Cookbook and FLTK::CheatSheet

Author
    Sanko Robinson <sanko@cpan.org> - http://sankorobinson.com/

License and Legal
    Copyright (C) 2008-2010 by Sanko Robinson <sanko@cpan.org>

    This program is free software; you can redistribute it and/or modify it
    under the terms of The Artistic License 2.0. See the LICENSE file
    included with this distribution or
    http://www.perlfoundation.org/artistic_license_2_0. For clarification,
    see http://www.perlfoundation.org/artistic_2_0_notes.

    When separated from the distribution, all original POD documentation is
    covered by the Creative Commons Attribution-Share Alike 3.0 License. See
    http://creativecommons.org/licenses/by-sa/3.0/us/legalcode. For
    clarification, see http://creativecommons.org/licenses/by-sa/3.0/us/.

$Id$


Changes
Version 0.532008 | 2011-07-02 23:10:36Z (Sat, 02 Jul 2011) | 1529294

    API Changes/Compatibility Information:
     * add_fd( fileno($fh), ... ) is back

Version 0.532008 | 2011-07-02 23:10:36Z (Sat, 02 Jul 2011) | 1529294

    Resolved Issues/Bug Fixes:
     * io watchers, idle watchers, timers, et al. now work correctly
     * Primitive FLTK::Rectangle can now be subclassed and extended to create
         complete custom widgets in Perl (example to come)
     * FLTK::Group->add( $widget ), et al. correctly return $widget

    API Changes/Compatibility Information:
     * io watchers, etc. return a blessed object which triggers destruction
         when needed to make life somewhat easier
     * FLTK::TabGroup->default_pager(0|1) uses built-in pager types
     * New Widgets/Types
       - TabGroupPager

Version 0.532007 | 2011-02-02 21:48:26Z (Wed, 02 Feb 2011) | 3d6d95a

    Resolved Issues/Bug Fixes:
     * $group->child( $i ) no longer crashes when $group has less than $i kids
     * FLTK::Window::first( ) correctly returns the first/primary window

    API Changes/Compatibility Information:
     * FLTK::Group->add( ... ), ->insert( ... ), and ->replace( ... ) now
         return the widget added, inserted, or used as a replacement. This
         allows code like my $box = $window->add( FLTK::Widget->( ... ) ); and
         is not compatible with the C++ API
     * New Widgets/Types
       - StatusBarGroup, ToggleButton, ToggleItem, ValueSlider
     * Add support for add_fd( ... ) and remove_fd( ... )
     * FLTK::GLWindow is no longer disabled on Win32 (try the example)

    Documentation/Sample Code/Test Suite/Metadata:
     * Updated repo URLs
     * Updated README
     * Moved basic POD from xs/FLTK.xs to lib/FLTK.pm
     * Allow systems to skip tests when Test::NeedsDisplay fails
     * Upgraded ask( ... ) and choice( ... ) to major C++/Perl API differences
     * New, simple IO watcher example (echo server)
     * New, simple add_fd( ... ) and remove_fd( ... ) tests
     * Final docs are generated on Build dist rather than Build install
     * Delay test plan until we know Test::NeedsDisplay is finished

    Notes:
     * Requires AF2 0.08323

Version 0.532006 | 2010-09-27 04:13:25Z (Mon, 27 Sep 2010) | 04ca288

    Resolved Issues/Bug Fixes:
     * Cleaned up general stupidity in FLTK::TabGroup->tab_positions()
     * Attempt to use the correct resource builder for MSVC
     * Fix doublefree when ...->destroy() is done manually on subclassed
         widgets

    Internal/Behavioral Changes:
     * Build system has been tweaked to avoid a bug in ExtUtils::ParseXS
     * Require Perl 5.10 (which I consider quite old) or better

    Documentation/Sample Code/Test Suite/Metadata:
     * New FLTK::Cookbook::TabGroup::Simple
     * Move from META.yml to META.json. The future is now.

Version 0.532005001 | 2010-04-17 14:25:12Z (Sat, 17 Apr 2010) | b9f4e5a

    API Changes/Compatibility Information:
     * New Widgets/Types
       - TextDisplay, TextDisplay::StyleTableEntry
     * Early FLTK::Theme support

    Meanwhile:
     * 開例 R. was born November 16th at 8:58am. She was 3.670kg and 52cm long
     * Requires Alien::FLTK2 now that Alien::FLTK only builds the 1.3.x branch

Version 0.531 | 2009-11-06 03:34:36Z (Fri, 06 Nov 2009) | 53908ec

    API Changes/Compatibility Information:
     * New Widgets/Types
       - PackedGroup, PixelType, Preferences, ProgressBar, RadioButton,
           RadioItem, RadioLightButton, Rectangle (base class of all Widgets),
           RepeatButton, ReturnButton, Scrollbar, ScrollGroup, SecretInput
           (for passwords), ShapedWindow, Slider, Symbol (.*Image base class),
           Valuator, ValueInput, and xpmImage
     * Support for fltk's file and directory chooser API
     * The variables $FLTK::ok, $FLTK::yes, $FLTK::no, $FLTK::cancel,
           $FLTK::message_window_label, $FLTK::message_window_timeout, and
           $FLTK::message_window_scrollable are now connected to their C++
           counterparts and may be imported with the :vars tag

    Resolved Issues/Bug Fixes:
     * FLTK::Menu->item() now (attempts to) return correctly blessed widgets

    Internal/Behavioral Changes:
     * The :default tag is no longer automatically exported

    Documentation/Sample Code/Test Suite:
     * Basic Rectangle tests
     * New example script for xpmImage
     * New FLTK::CheatSheet (incomplete)
     * First few recipes in the FLTK::Cookbook

    Notes:
     * The FLTK project is just getting started so now's a great time to join

    Meanwhile:
     * Ian G. was born October 23rd at 1:22pm. He was 8lbs 14oz and 22in long

Version 0.530 | 2009-10-05 00:42:52Z (Mon, 05 Oct 2009) | 3b70f43

    API Changes/Compatibility Information:
     * ...see below

    Resolved Issues/Bug Fixes:
     * ...see below

    Internal/Behavioral Changes:
     * It actually exists

    Documentation/Sample Code/Test Suite:
     * ...see above

    Notes:
     * The FLTK project is just getting started so now's a great time to join

_ -.-. .... .- -. --. . ... _________________________________________
For more information, see the commit log:
    http://github.com/sanko/perl-fltk/commits/master/

$Ver$ from git $Rev$
$Date$
$Url$


lib/FLTK.pm
package FLTK;
{

=pod

=for license Artistic License 2.0 | Copyright (C) 2009-2011 by Sanko Robinson

=for author Sanko Robinson <sanko@cpan.org> - http://sankorobinson.com/

=for abstract Perl bindings to the Fast Light Toolkit

=for git $Id$

=cut
    use strict;
    use warnings;
    our $MAJOR = 532; our $MINOR = 9; our $DEV = 0; our $VERSION = sprintf('%1.3f%03d' . ($DEV ? (($DEV < 0 ? '' : '_') . '%03d') : ('')), $MAJOR / 1000, $MINOR, abs $DEV);
    use XSLoader;
    use vars qw[@EXPORT_OK @EXPORT %EXPORT_TAGS];
    use Exporter qw[import];

    #
    our $NOXS ||= $0 eq __FILE__;    # for testing
    XSLoader::load 'FLTK', $VERSION
        if !$FLTK::NOXS;             # Fills %EXPORT_TAGS on BOOT

    #
    @EXPORT_OK = sort map { @$_ = sort @$_; @$_ } values %EXPORT_TAGS;
    $EXPORT_TAGS{'all'} = \@EXPORT_OK;    # When you want to import everything
    @{$EXPORT_TAGS{'style'}}              # Merge these under a single tag
        = sort map { defined $EXPORT_TAGS{$_} ? @{$EXPORT_TAGS{$_}} : () }
        qw[box font label]
        if 1 < scalar keys %EXPORT_TAGS;
    @EXPORT  # Export these tags (if prepended w/ ':') or functions by default
        = sort map { m[^:(.+)] ? @{$EXPORT_TAGS{$1}} : $_ }
        qw[:style :default]
        if 0 && keys %EXPORT_TAGS > 1;
}
1;

=pod

=encoding utf8

=head1 NAME

FLTK - Perl bindings to the 2.0.x branch of the Fast Light Toolkit

=head1 Synopsis

=for markdown {%highlight perl linenos%}

    use strict;
    use warnings;
    use FLTK qw[:style];

    my $win = FLTK::Window->new(300, 180);
    my $box = $win->add(FLTK::Widget->new(20, 40, 260, 100, 'Hello, World!'));
    $box->box(UP_BOX);
    $box->labelfont(HELVETICA_BOLD_ITALIC);
    $box->labelsize(36);
    $box->labeltype(SHADOW_LABEL);
    $win->show();
    exit FLTK::run();

=for markdown {%endhighlight%}

=head1 Description

FLTK is a graphical user interface toolkit for X (UNIX®), Microsoft® Windows®,
OS/X, and several other platforms. FLTK provides modern GUI functionality
without the bloat and supports 3D graphics via OpenGL® and its built-in GLUT
emulation.

This module, L<FLTK|FLTK>, exposes bindings to the experimental 2.0.x branch
of the Fast Light Toolkit.

=head1 See Also

L<FLTK::Notes|FLTK::Notes>, L<FLTK::Basics|FLTK::Basics>,
L<FLTK::Cookbook|FLTK::Cookbook>, and L<FLTK::CheatSheet|FLTK::CheatSheet>

=head1 Author

Sanko Robinson <sanko@cpan.org> - http://sankorobinson.com/

CPAN ID: SANKO

=head1 License and Legal

Copyright (C) 2008-2010 by Sanko Robinson <sanko@cpan.org>

This program is free software; you can redistribute it and/or modify it under
the terms of
L<The Artistic License 2.0|http://www.perlfoundation.org/artistic_license_2_0>.
See the F<LICENSE> file included with this distribution or
L<notes on the Artistic License 2.0|http://www.perlfoundation.org/artistic_2_0_notes>
for clarification.

When separated from the distribution, all original POD documentation is
covered by the
L<Creative Commons Attribution-Share Alike 3.0 License|http://creativecommons.org/licenses/by-sa/3.0/us/legalcode>.
See the
L<clarification of the CCA-SA3.0|http://creativecommons.org/licenses/by-sa/3.0/us/>.

=for git $Id$

=cut
